app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt

```diff
@@ -373,3 +473,75 @@ fun currentFriendStreak(lastLoginDateString: String?, currentStreak: Long): Long
   }
   return -1L // No last login date recorded
 }
+/**
+ * Composable function that represents a single friend request item in the list.
+ *
+ * It displays:
+ * - The requester's profile picture, name, and bio.
+ * - Buttons to accept or decline the friend request.
+ * - Handles user interactions and updates the state through the ViewModel.
+ *
+ * @param friendRequest The [UserProfile] object representing the user who sent the request.
+ * @param userProfileViewModel The [UserProfileViewModel] that handles accepting or declining the
+ *   request.
+ */
+@Composable
+fun FriendRequestItem(friendRequest: UserProfile, userProfileViewModel: UserProfileViewModel) {
+  Surface(
+      modifier =
+          Modifier.fillMaxWidth()
+              .padding(horizontal = AppDimensions.smallPadding)
+              .clip(RoundedCornerShape(AppDimensions.roundedCornerRadius))
+              .testTag("friendRequestItem#${friendRequest.uid}"),
+      color = AppColors.LightPurpleGrey,
+      shadowElevation = AppDimensions.elevationSmall) {
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(AppDimensions.paddingMedium),
+            verticalAlignment = Alignment.CenterVertically) {
+              // Friend's Profile Picture
+              ProfilePicture(
+                  profilePictureUrl = friendRequest.profilePic,
+                  onClick = { /* Optionally, show enlarged picture */})
+              Spacer(modifier = Modifier.width(AppDimensions.smallWidth))
+              Column(modifier = Modifier.weight(1f)) {
+                // Friend's Name
+                Text(
+                    text = friendRequest.name,
+                    style = MaterialTheme.typography.titleMedium,
+                    modifier =
+                        Modifier.padding(bottom = AppDimensions.smallPadding)
+                            .testTag("friendRequestName#${friendRequest.uid}"))
+                // Friend's Bio
+                Text(
+                    text = friendRequest.bio ?: "No bio available",
+                    style = MaterialTheme.typography.bodySmall,
+                    color = AppColors.secondaryTextColor,
```

Comment:
The color should be retrieved using `MaterialTheme.colorScheme` as discussed earlier.